src/main/daml/Daml/Finance/Instrument/Swap/Asset/DistributionRule.daml

```diff
@@ -0,0 +1,93 @@
+-- Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+module Daml.Finance.Instrument.Swap.Asset.DistributionRule where
+
+import DA.List (dedup, head)
+import DA.Map(fromList)
+import Daml.Finance.Interface.Instrument.Base.Instrument qualified as BaseInstrument (exerciseInterfaceByKey)
+import Daml.Finance.Interface.Instrument.Swap.Asset.Factory qualified as AssetSwapFactory (Create(..), I(..))
+import Daml.Finance.Interface.Instrument.Swap.Asset.Instrument qualified as Asset (GetView(..), I)
+import Daml.Finance.Interface.Lifecycle.Rule.Lifecycle qualified as Lifecycle (Evolve(..), I, View(..))
+import Daml.Finance.Interface.Types.Common.Types (Id(..), Parties)
+import Daml.Finance.Interface.Util.Common (scale)
+import Daml.Finance.Lifecycle.Effect (Effect(..))
+import Daml.Finance.Lifecycle.Event.Distribution qualified as Distribution (Event(..))
+
+-- | Rule contract that defines the distribution of units of an instrument for each unit of a target
```

Comment:
Shall we be more specific and say that it's a rule used to process a `Distribution` event targeting an underlying for an asset swap?